lib/neo/dci/context.rb

```diff
@@ -1,39 +1,41 @@
+require "ruby2_keywords"
+
 module Neo
   module DCI
     class Context
       class << self
         private :new
-      end
 
-      attr_accessor :callback
+        def callbacks(*args)
+          @callbacks ||= []
+          @callbacks = args unless args.empty?
+          @callbacks
+        end
 
-      def self.callbacks(*args)
-        @callbacks ||= []
-        @callbacks = args unless args.empty?
-        @callbacks
-      end
+        ruby2_keywords def call(*args, &block)
```

Comment:
The changes in this file look more than they are.

The only important change is to mark the `call` method with `ruby2_keywords` to make a compatible delegation that works on Ruby <=2.6, 2.7 and Ruby 3:

Ruby <= 2.6 does not handle the new delegation style correctly, so the old-style delegation must be used.
`ruby2_keywords` allows you to run the old style even in Ruby 2.7 and 3.0.

More informations:
https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

Discussion about adding `ruby2_keywords` semantics by default in 2.7.1:
https://bugs.ruby-lang.org/issues/16463

We need to remove this method if we no longer support Ruby versions < 2.7.